pg_search/tests/bm25_search.rs

```diff
@@ -1065,3 +1065,225 @@ fn index_size(mut conn: PgConnection) {
     // Ensure the size is greater than zero, meaning the index has been created
     assert!(size > 0);
 }
+
+#[rstest]
+fn separate_nested_json_docs(mut conn: PgConnection) {
+    r#"
+    CREATE TABLE customers (
+        id SERIAL PRIMARY KEY,
+        name TEXT,
+        crm_data JSONB
+    );
+
+    INSERT INTO customers (name, crm_data)
+    VALUES 
+    ('Customer A', '[{
+            "interaction": "call",
+            "details": {
+                "subject": "Welcome Call",
+                "date": "2023-09-01"
+            }
+        }, {
+            "interaction": "email",
+            "details": {
+                "subject": "Goodbye Email",
+                "date": "2023-09-02"
+            }
+        }]'::JSONB
+    );
+    "#
+    .execute(&mut conn);
+
+    r#"
+    CALL paradedb.create_bm25(
+        index_name => 'customers',
+        table_name => 'customers',
+        key_field => 'id',
+        text_fields => paradedb.field('name'),
+        json_fields => paradedb.field('crm_data')
+    );
+    "#
+    .execute(&mut conn);
+
+    // Test search with nested JSON that does not match both conditions within the same object
+    let rows: Vec<(i32, String)> = r#"
+    SELECT id, name
+    FROM customers.search('crm_data.interaction:call AND crm_data.details.subject:Goodbye');
+    "#
+    .fetch_collect(&mut conn);
+    assert_eq!(rows.len(), 0); // Customer A should NOT be found
+
+    // Test search with nested JSON that matches both conditions in the same object
+    let rows: Vec<(i32, String)> = r#"
+    SELECT id, name
+    FROM customers.search('crm_data.interaction:email AND crm_data.details.subject:Goodbye');
+    "#
+    .fetch_collect(&mut conn);
+    assert_eq!(rows, vec![(1, "Customer A".to_string())]); // Customer A should be found
+
+    // Test search with a JSON object that does not match any conditions
+    let rows: Vec<(i32, String)> = r#"
+    SELECT id, name
+    FROM customers.search('crm_data.interaction:sms AND crm_data.details.subject:Reminder');
+    "#
+    .fetch_collect(&mut conn);
+    assert_eq!(rows.len(), 0); // No customers should be found
+
+    // Test search with only one condition
+    let rows: Vec<(i32, String)> = r#"
+    SELECT id, name
+    FROM customers.search('crm_data.interaction:call');
+    "#
+    .fetch_collect(&mut conn);
+    assert_eq!(rows, vec![(1, "Customer A".to_string())]); // Customer A should be found
+}
+
+#[rstest]
+fn larger_dataset_nested_json(mut conn: PgConnection) {
+    // Create the table and insert dataset
+    r#"
+    CREATE TABLE customers (
+        id SERIAL PRIMARY KEY,
+        name TEXT,
+        crm_data JSONB
+    );
+
+    INSERT INTO customers (name, crm_data)
```

Comment:
Suggestion: We can use my new `create_bm25_test_table` function to add a new `customers` table type, so that we'll be able to use the same table in the documentation and tests.